### PR TITLE
test(atomic): refactor category facet tests

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-633

Basically 3 sections for tests:

**Selectors:** similar to before instead that selectors are not only CSS selectors string values, but they actually return the elements e.g. `cy.get(....).find(...)` also contains some other useful values, changing the selection is now trivial. No more alias issues as well (alias can break setup if they are not found)

**Assertions:** file purely containing assert method, which contain the "it" with the description. This is easily modifiable & maintainable, as it leverages TS and prevents hardcoding.

**Test file:** Simply sets up the different conditions and call assertions. For assertions which simply check rendering elements and contain no interaction, I use `before` instead of `beforeEach`, this SPEEDS UP the tests a lot.

### CODE REVIEW

File: packages/atomic/cypress/integration/facets/category-facet-assertions.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+import {ResultListSelectors} from '../result-list-selectors';
+import {
+  hierarchicalField,
+  CategoryFacetSelectors,
+  categoryFacetComponent,
+  BreadcrumbSelectors,
+} from './category-facet-selectors';
+
+// TODO: move util out
+function should(should: boolean) {
```
Comment (samisayegh) [inline packages/atomic/cypress/integration/facets/category-facet-assertions.ts:13]: I feel functions like this one are starting to enter the realm of over-engineering?

Personally, I would keep the `it` blocks and their descriptions explicitly written out in the test files. This would provide flexibility to express nuance (examples, edge cases) in the descriptions. It would also avoid having utility `should` functions.

The actual assertions can remain in this file though, reused inside multiple `it` blocks.

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/cypress/integration/facets/category-facet-assertions.ts (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/cypress/integration/facets/category-facet-assertions.ts:13]: I chuckled when writing this method, but it prevented me from rewriting the descriptions, which is one of the main purpose of this PR: to minimize repetition sand maximize consistency & reusability of assertions. 

Having the `it` description & the assertions in 2 different places would require checking back and forth to make sure the "assertion code" fits with the description, here, they go together.

I rather use a trivial util method than manually writing it out every time or copy-pasting. For example I would have to write out `should (or not) display a "All Categories" button` 7 times in this test suite only. I find Cypress tests have way more repetitions than when writing UTs, conditions have to be checked and rechecked many times. 

> provide flexibility to express nuance (examples, edge cases)

Most of the context & nuances are expressed using the describe blocks, which prepare the interface for the assertions. This pattern kind of enforces the good practice of writing 1 check/expect/assert per "it" block. Then as many assert methods as needed can then be written out, along with their it block.

Comment (olamothe) [reply packages/atomic/cypress/integration/facets/category-facet-assertions.ts:12]: I'm fine if it "helps reduce some repetition", but, yeah.... repetition is not always bad, you know ;) 

It's boring to write, sure, but it's easier to read.

Anyway, as long as we don't "force this" for all tests, if it makes sense for a particular test suite 👍

File: packages/atomic/cypress/integration/facets/category-facet.cypress.ts [hunk 2]
```diff
@@ -1,617 +1,372 @@
...
+  describe('with default settings', () => {
+    function setupWithDefaultSettings() {
+      setupCategoryFacet();
+      cy.wait(RouteAlias.search);
+    }
+
+    describe('verify rendering', () => {
+      before(setupWithDefaultSettings);
+
+      CategoryFacetAssertions.assertAccessibility();
```
Comment (olamothe) [inline packages/atomic/cypress/integration/facets/category-facet.cypress.ts:54]: Another approach (I'm not saying it would be better) would be to inverse the logic.

Here, we do "for a given setup, test/assert those N different features are okay". 

What if it was "for a given feature, create N different setup, and assert the given feature is okay".

eg:

```
for (const setup of [()=> setupWithDefaultSeting(), ()=> setupGoDeeperOneLevel(), ()=> setupSelectFirstParent(), .....]) {
  setup();
 CategoryFacetAssertions.assertFoo();
} 
```

Just food for thoughts.

**Review by olamothe — APPROVED** [review]

File: packages/atomic/cypress/integration/facets/category-facet.cypress.ts (hunk 2 above)
Comment (lvu285) [reply packages/atomic/cypress/integration/facets/category-facet.cypress.ts:54]: I still prefer the original way, as later we might need to reuse these small tests for E2E tests on index.html/or client similar environment.
